feat(discover): Give a tip when an IN condition could be used

wmak opened the pull request:
- This checks that there's at least one OR condition between two
  identical fields, and returns a tip on the events meta that could help
  users write better queries

wmak committed:
feat(discover): Give a tip when an IN condition could be used

- This checks that there's at least one OR condition between two
  identical fields, and returns a tip on the events meta that could help
  users write better queries

priscilawebdev commented on tests/snuba/api/endpoints/test_organization_events_v2.py:
```diff
@@ -10962,6 +10962,20 @@ def test_tag_that_looks_like_aggregate(self):
+        assert response.status_code == 200, response.content
+        meta = response.data["meta"]
+        assert meta["tips"] == {
+            "query": "Did you know you can replace chained or conditions like `field:a OR field:b OR field:c` with `field:[a,b,c]`",
```
wouldn't it be better if we handle the phrases on the frontend using one of our translation functions?

wmak replied:
Do we do anything like that for our 400 errors?

edwardgou-sentry reviewed (approved):
one inconsistency I noticed is that `title:a or title:b or event.type:transaction` won't trigger the hint but `event.type:transaction or title:a or title:b` does trigger the hint.

not a blocker though, everything else lgtm
edit: fixed query example

wmak committed:
ref: Moving message to constants, and fixing or chain bug

- Need to check if the right side is already a chain of ORs and if the
  leftmost condition is the matching field or not

wmak committed:
fix: typing
